File: joss/comparisons/cat-3-null-missing-data/26-sort-nullable-columns/example-sort-nullable-columns.R
# Error Class 26: Sorting on Nullable Columns
#
# R silently sorts NA values to the end with arrange().
# No warning is given about the implicit NA placement.

library(tidyverse)

labs <- tibble(
  patient_id = c("P001", "P002", "P003"),
  result_value = c(100, NA, 50)
)

# SILENT: NA rows sorted to end — no warning
sorted_df <- labs %>% arrange(result_value)
print(sorted_df)
# P003: 50, P001: 100, P002: NA — NA silently at end

File: joss/comparisons/cat-3-null-missing-data/26-sort-nullable-columns/example-sort-nullable-columns.py
"""
Error Class 26: Sorting on Nullable Columns

Python silently sorts NaN values to the end by default.
The na_position parameter exists but is optional — the default
behavior is implicit and undocumented at the point of use.
"""
import pandas as pd
import numpy as np

labs = pd.DataFrame({
    "patient_id": ["P001", "P002", "P003"],
    "result_value": [100, np.nan, 50],
})

# SILENT: NaN rows sorted to end — no warning about implicit behavior
sorted_df = labs.sort_values("result_value")
print(sorted_df)
# P003: 50, P001: 100, P002: NaN — NaN silently at end
